pkg/reconciler/contour/lister.go

```diff
@@ -63,35 +71,33 @@ func (l *lister) ListProbeTargets(ctx context.Context, ing *v1alpha1.Ingress) ([
 		urls := make([]*url.URL, 0, hosts.Len())
 		for _, host := range hosts.UnsortedList() {
 			urls = append(urls, &url.URL{
-				Scheme: "http",
+				Scheme: scheme,
 				Host:   host,
 			})
 		}
 
-		// TODO(mattmoor): Perhaps key off of whether HTTP is enabled?
-		portName, err := network.NameForPortNumber(service, 80)
+		portName, err := network.NameForPortNumber(service, port)
```

Comment:
need to make sure `port` is int32 here. so I think it's better to cast it to int32 here instead of on line #48

Reply:
`port` gets defined as int32 on #48. The untyped `443` constant on #51 is implicitly converted to int32 on assignment[1].

[1] https://golang.org/ref/spec#Representability